### PR TITLE
Add multipart parsing placeholders

### LINKED ISSUE #32: Reactive parser for multipart.

### LINKED ISSUE #46: RqMultipart.java:38-41: Implement this class. It should...
The puzzle `32-7952a3d2` from #32 has to be resolved: 

https://github.com/artipie/http/blob/4f72e29e24654052ebd667db6bb9f537b167b693/src/main/java/com/artipie/http/rq/RqMultipart.java#L38-L41

The puzzle was created by @g4s8 on 20-Feb-20. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### LINKED ISSUE #47: RqMultipart.java:69-71: Add secondary constructor...
The puzzle `32-5080620a` from #32 has to be resolved: 

https://github.com/artipie/http/blob/4f72e29e24654052ebd667db6bb9f537b167b693/src/main/java/com/artipie/http/rq/RqMultipart.java#L69-L71

The puzzle was created by @g4s8 on 20-Feb-20. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by Sammers21**
This PR includes:
* A test, in which multipart parser is used for parsing real multipart requests
* 2 puzzles for implementing Processor interface
* Placeholders for the processor interface

### CODE REVIEW

**Comment by paulodamaso** [comment]
@g4s8 please assign me

**Comment by Sammers21** [comment]
@paulodamaso, we have zerocrat for issue assigning. Stop asking for manual assign.

**Comment by 0crat** [comment]
This pull request [#50](https://github.com/artipie/http/issues/50) is assigned to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/165f7173-992a-49c8-8c7d-499aa1e81953); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by fanifieiev** [comment]
@0crat refuse

**Comment by paulodamaso** [comment]
@Sammers21 Thanks for the tip

**Comment by paulodamaso** [comment]
@g4s8 I'm available, please assign me

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/artipie/http/issues/50#issuecomment-589721633))

@fanifieiev The user @fanifieiev[/z](https://www.0crat.com/u/fanifieiev) resigned from [#50](https://github.com/artipie/http/issues/50), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/artipie/http/issues/50#issuecomment-589721633))

@fanifieiev Job refused in 7 hours - no penalty, see [§6](http://www.zerocracy.com/policy.html#6)

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) not enough funds available in the project, can't set budget of job [#50](https://github.com/artipie/http/issues/50), see [§21](http://www.zerocracy.com/policy.html#21); @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) will get no money on completion; in order to fix that, add funds to the project and assign the job again

**Comment by 0crat** [comment]
@paulodamaso[/z](https://www.0crat.com/u/paulodamaso) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by paulodamaso** [comment]
@Sammers21 Is this PR ready to review?

**Comment by Sammers21** [comment]
@paulodamaso , Yes, it is ready

File: src/test/java/com/artipie/http/rq/MpTest.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            );
+            return CompletableFuture.completedFuture(null);
+        };
+        final int port = this.rndPort();
+        final VertxSliceServer server = new VertxSliceServer(vertx, slice, port);
+        server.start();
+        final Path resolve = dir.resolve("text.txt");
+        Files.write(resolve, "Hello worrrrld!!!".getBytes());
+        final WebClient web = WebClient.create(vertx);
+        web.post(port, MpTest.LOCALHOST, "/hello")
```
Comment (paulodamaso) [inline src/test/java/com/artipie/http/rq/MpTest.java]: @Sammers21 You can supress `web` variable here, it is just used once.

File: src/test/java/com/artipie/http/rq/MpTest.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+import java.nio.file.Path;
+import java.util.HashSet;
+import java.util.concurrent.CompletableFuture;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+/**
+ * Multipart parser test.
+ * @todo #32:60min Enable this test.
```
Comment (paulodamaso) [inline src/test/java/com/artipie/http/rq/MpTest.java]: @Sammers21 This test doesn't look complete; you should test somehow if the multipart form is correctly parsed, not just execute it and expect that no error is thrown. Also, I think that we don't need a puzzle just for enabling those tests; we could add this information in the puzzles related to finishing `Mp` implementation

File: src/main/java/com/artipie/http/rq/Mp.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+ *  Subscribe is a part of publisher contract. Parts are emmited in a way, similar to stream parser,
+ *  but with an attention to headers.
+ * @todo #32:120min Implement Subscriber part.
+ *  On subscribe is called when upstream wants to propagate elements to us. We should take it and
+ *  use it.
+ * @since 0.4
+ * @checkstyle ConstantUsageCheck (500 lines)
+ */
+@SuppressWarnings({"PMD.AvoidDuplicateLiterals", "PMD.UnusedPrivateField", "PMD.SingularField"})
+public final class Mp implements Processor<ByteBuffer, Part> {
```
Comment (paulodamaso) [inline src/main/java/com/artipie/http/rq/Mp.java]: @Sammers21 You could call this class `Multipart`, I think that it becomes clearer to the reader

**Comment by paulodamaso** [comment]
@Sammers21 Please take a look at the comments

File: src/main/java/com/artipie/http/rq/Mp.java (hunk 3 above)
Comment (Sammers21) [reply src/main/java/com/artipie/http/rq/Mp.java]: @paulodamaso , agree

File: src/test/java/com/artipie/http/rq/MpTest.java (hunk 2 above)
Comment (Sammers21) [reply src/test/java/com/artipie/http/rq/MpTest.java]: @paulodamaso , looks reasonable

Comment (Sammers21) [reply src/test/java/com/artipie/http/rq/MpTest.java]: @paulodamaso , done

**Comment by Sammers21** [comment]
@paulodamaso , please, look at this pull request once again.

File: src/test/java/com/artipie/http/rq/MpTest.java (hunk 2 above)
Comment (paulodamaso) [reply src/test/java/com/artipie/http/rq/MpTest.java]: @Sammers21 I still don't see where the tests assure that the `Multipart` parser works correctly; you should test it by providing a set of known values to Multipart and checking the generated output.

**Comment by Sammers21** [comment]
@paulodamaso , review the pull request once again please

**Comment by Sammers21** [comment]
@g4s8 , this pull request is blocked by #84

**Comment by Sammers21** [comment]
@paulodamaso, please, look at this pull request once again. I have just added an assertion that multipart is parsed correctly.

File: src/test/java/com/artipie/http/rq/MultipartTest.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                RsStatus.OK,
+                new HashSet<>(zero),
+                Flowable.empty()
+            );
+            return CompletableFuture.completedFuture(null);
+        };
+        final int port = this.rndPort();
+        final VertxSliceServer server = new VertxSliceServer(vertx, slice, port);
+        server.start();
+        final Path resolve = dir.resolve("text.txt");
```
Comment (paulodamaso) [inline src/test/java/com/artipie/http/rq/MultipartTest.java]: @Sammers21 Great, could you please just inline `slice`and `port` into `server` variable instantiation?

Comment (Sammers21) [reply src/test/java/com/artipie/http/rq/MultipartTest.java]: @carlosmiranda, port can't be inlined

**Comment by paulodamaso** [comment]
@g4s8 please merge

**Comment by 0crat** [comment]
Job `gh:artipie/http#50` is not assigned, can't get performer

**Comment by paulodamaso** [comment]
@g4s8 I could'nt find the payment for this one, please take a look

**Comment by g4s8** [comment]
@0crat pay 15min @paulodamaso

**Pull request merged**